UofT-Frosh-Orientation/orientation-website
# 627 optimize image delivery

## **Contribution:**

### **Issue:**

Closes #627  

### **Accomplishments:**

- downloading big images locally (mainly the home page slideshow)
  - using webp format 
  - using progressive loading to help with loading the large images from slideshow
- note for later, sponsorship images also take a long loading time, will need to convert those before adding them
- adding lazy loading for other larger images on home page & about page
- otherwise, other logos and graphics didn't take too long to load and sizes were small

### **Visuals:** 

Same visuals as before

## **Details:**

### **Expected Behavior:**

- performance is better (hopefully?)
- slideshow should show blurry image first and then a very high quality image once it loads

### **Testing Steps:**

Steps required to get expected behavior.

- load frontend and check out the home page, mainly the slideshow

### **Complications:**

- addressed in resources

### **Resources:**

Any additional resources used outside this in issue.

- [compressing files](https://www.xconvert.com/compress-webp) 
- [progressive image loading](https://blog.logrocket.com/progressive-image-loading-react-tutorial/)
- [webp converter](https://cloudconvert.com/webp-converter)
